### PR TITLE
Add a new geometry example using different geoms in the same mesh.

### PR DESCRIPTION
**Pull request by holke**
**_Describe your changes here:_**

The new example uses the cylinder and sincos geometries.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] New Datatypes are added to t8indent_custom_datatypes.txt
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] New tests use the Google Test framework
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [x] There is a at least an example covering the new code, better even a wiki article or tutorial
- [x] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: example/geometry/t8_example_geometries.cxx [hunk 1]
```diff
@@ -511,6 +513,23 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
...
+      ("Creating uniform level %i forest with a cylinder and a sinus cosinus geometry.\n",
+       level);
+    /* Cylinder geometry on tree 0. Sincos geometry on tree 1. */
+    geometry = new t8_geometry_cylinder;
+    geometry_sincos = new t8_geometry_sincos;
+    t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_QUAD);
+    /* Tree 0 is connecte to itself to form a cylinder */
+    t8_cmesh_set_join (cmesh, 0, 0, 0, 1, 0);
+    t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_QUAD);
+    t8_cmesh_register_geometry (cmesh, geometry_sincos);
```
Comment (sandro-elsweijer) [inline example/geometry/t8_example_geometries.cxx:527]: We should comment, that both geometries have to be registered and that the cylinder geometry is registered later on

File: example/geometry/t8_example_geometries.cxx [hunk 2]
```diff
@@ -511,6 +513,23 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
...
     break;
+  case T8_GEOM_TWO_GEOMETRIES:
+    t8_global_productionf
+      ("Creating uniform level %i forest with a cylinder and a sinus cosinus geometry.\n",
+       level);
+    /* Cylinder geometry on tree 0. Sincos geometry on tree 1. */
+    geometry = new t8_geometry_cylinder;
+    geometry_sincos = new t8_geometry_sincos;
+    t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_QUAD);
+    /* Tree 0 is connecte to itself to form a cylinder */
```
Comment (sandro-elsweijer) [inline example/geometry/t8_example_geometries.cxx:524]:
```suggestion
    /* Tree 0 is connected to itself to form a cylinder */
```

File: example/geometry/t8_example_geometries.cxx [hunk 3]
```diff
@@ -511,6 +513,23 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
       snprintf (vtuname, BUFSIZ, "forest_moebius_lvl_%i", level);
     }
     break;
+  case T8_GEOM_TWO_GEOMETRIES:
+    t8_global_productionf
+      ("Creating uniform level %i forest with a cylinder and a sinus cosinus geometry.\n",
```
Comment (sandro-elsweijer) [inline example/geometry/t8_example_geometries.cxx:518]:
```suggestion
      ("Creating uniform level %i forest with a cylinder and a sine cosine geometry.\n",
```

File: example/geometry/t8_example_geometries.cxx [hunk 4]
```diff
@@ -471,6 +472,7 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
   t8_cmesh_t          cmesh;
   char                vtuname[BUFSIZ];
   t8_geometry_c      *geometry;
+  t8_geometry_c      *geometry_sincos;
```
Comment (sandro-elsweijer) [inline example/geometry/t8_example_geometries.cxx:475]:
```suggestion
  t8_geometry_c      *geometry_sincos; /* Used for T8_GEOM_TWO_GEOMETRIES*/
```

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
Some minor improvements

File: example/geometry/t8_example_geometries.cxx [hunk 5]
```diff
@@ -511,6 +513,23 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
...
+      ("Creating uniform level %i forest with a cylinder and a sine cosine geometry.\n",
+       level);
+    /* Cylinder geometry on tree 0. Sincos geometry on tree 1. */
+    geometry = new t8_geometry_cylinder;
+    geometry_sincos = new t8_geometry_sincos;
+    t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_QUAD);
+    /* Tree 0 is connected to itself to form a cylinder */
+    t8_cmesh_set_join (cmesh, 0, 0, 0, 1, 0);
+    t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_QUAD);
+    t8_cmesh_register_geometry (cmesh, geometry_sincos);
```
Comment (holke) [inline example/geometry/t8_example_geometries.cxx:527]:
```suggestion
    t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_QUAD);
    /* Note that we have to register both geometries to the cmesh. The cylinder geometry is
     * stored in the "geometry" pointer and registered later right before the cmesh is committed. */
    t8_cmesh_register_geometry (cmesh, geometry_sincos);
```

File: example/geometry/t8_example_geometries.cxx [hunk 6]
```diff
@@ -511,6 +513,25 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
...
+       level);
+    /* Cylinder geometry on tree 0. Sincos geometry on tree 1. */
+    geometry = new t8_geometry_cylinder;
+    geometry_sincos = new t8_geometry_sincos;
+    t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_QUAD);
+    /* Tree 0 is connected to itself to form a cylinder */
+    t8_cmesh_set_join (cmesh, 0, 0, 0, 1, 0);
+    t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_QUAD);
+    /* Note that we have to register both geometries to the cmesh. The cylinder geometry is
+     * stored in the "geometry" pointer and registered later right before the cmesh is committed. */
```
Comment (holke) [inline example/geometry/t8_example_geometries.cxx:528]:
```suggestion
     * stored in the "geometry" pointer and registered later, right before the cmesh is committed. */
```

**Comment by holke** [comment]
I am done, handing this back to you.

File: example/geometry/t8_example_geometries.cxx (hunk 1 above)
Comment (holke) [reply example/geometry/t8_example_geometries.cxx:529]: done

File: example/geometry/t8_example_geometries.cxx [hunk 7]
```diff
@@ -471,6 +472,7 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
   t8_cmesh_t          cmesh;
   char                vtuname[BUFSIZ];
   t8_geometry_c      *geometry;
+  t8_geometry_c      *geometry_sincos; /* Used for T8_GEOM_TWO_GEOMETRIES*/
```
Comment (holke) [inline example/geometry/t8_example_geometries.cxx:475]:
```suggestion
  t8_geometry_c      *geometry_sincos; /* Used for T8_GEOM_TWO_GEOMETRIES */
```

Comment (holke) [reply example/geometry/t8_example_geometries.cxx:475]: Fix indentation?

File: example/geometry/t8_example_geometries.cxx [hunk 8]
```diff
@@ -471,6 +472,7 @@ t8_analytic_geom (int level, t8_example_geom_type geom_type)
   t8_cmesh_t          cmesh;
   char                vtuname[BUFSIZ];
   t8_geometry_c      *geometry;
+  t8_geometry_c      *geometry_sincos; /* Used for T8_GEOM_TWO_GEOMETRIES */
```
Comment (holke) [inline example/geometry/t8_example_geometries.cxx:475]:
```suggestion
  /* geoemtry_sincos is used for T8_GEOM_TWO_GEOMETRIES */
  t8_geometry_c      *geometry_sincos; 
```

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**